### ISSUE TITLE
JADC got the big blind twice due to new player motion code

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#290]
odbphatmackakabig got moved from a table with 7 players to our table where we only had 5. There logically was only one place where he could go, which was between JADC, who had just paid the big blind and me, who had just paid the small blind.

I should have gotten the button, which I did, odbphatmackakabig should have had to sit out (instead he paid the small blind) and JADC should have gotten the small blind.

I think this just shows that my implementation of #247 was flawed. I'll look at it when I'm awake tomorrow, but probably after I've exercised.

```
Dealing Pot-Limit Binglaha

deadhead blinds 2000
JADC blinds 4000
1     JADC      2  1850 4000  [  ] [  ] [  ] [  ]
2   > tombayz   0 20150    0  [  ] [  ] [  ] [  ]
3     nnr13     0 24750    0  [  ] [  ] [  ] [  ]
4 B   Meeeee2   2 11250    0  [  ] [  ] [  ] [  ]
5     deadhead  2 46000 2000  [  ] [  ] [  ] [  ]
                My cards are: [Kd] [6c] [8c] [8s]
deadhead: gg
tombayz folds
nnr13: gg grizz
nnr13 folds
grizz: ty
Meeeee2 folds
It is 2000 to you
deadhead calls
JADC checks
1     JADC      2  1850 4000  [  ] [  ] [  ] [  ]
2    [tombayz]  0 20150[   0] 
3    [nnr13]    0 24750[   0] 
4 B  [Meeeee2]  2 11250[   0] 
5   > deadhead  2 44000 4000  [  ] [  ] [  ] [  ]
Board: 5s 8d Tc
                My cards are: [Kd] [6c] [8c] [8s]
It is 0 to you
deadhead bets 4000
JADC asked for and gets 60 more seconds
odbphatmackakabig has come from Table 277
JADC will be turning in in 2 lammers before the next hand
JADC calls and is all-in
Meeeee2 rolled a 4
1   > JADC               2     0 5850  [  ] [  ] [  ] [  ]
2    [tombayz]           0 20150[   0] 
3    [nnr13]             0 24750[   0] 
4 B  [Meeeee2]           2 11250[   0] 
5     deadhead           2 40000 8000  [  ] [  ] [  ] [  ]
6    [odbphatmackakabig] 1  5750[   0] 
Board: 5s 8d Tc 7h
                         My cards are: [Kd] [6c] [8c] [8s]
1   > JADC               2     0 5850  [  ] [  ] [  ] [  ]
2    [tombayz]           0 20150[   0] 
3    [nnr13]             0 24750[   0] 
4 B  [Meeeee2]           2 11250[   0] 
5     deadhead           2 40000 8000  [  ] [  ] [  ] [  ]
6    [odbphatmackakabig] 1  5750[   0] 
Board: 5s 8d Tc 7h Th
                         My cards are: [Kd] [6c] [8c] [8s]
Uncalled 2150 returned to deadhead
deadhead  Kd 6c 8c 8s
    High  Full House (Eights over Tens)   8c 8d 8s Th Tc
     Low  none
JADC      5d 7d 2h 9h
    High  Two Pair (Tens and Sevens)      Tc Th 7h 7d 9h
     Low  none
deadhead  wins 11700 for High

JADC redeemed a lammer for 5000

JADC redeemed a lammer for 5000

Dealing Pot-Limit Binglaha

odbphatmackakabig blinds 2000
JADC blinds 4000
1     JADC               0  6000 4000  [  ] [  ] [  ] [  ]
2   > tombayz            0 20150    0  [  ] [  ] [  ] [  ]
3     nnr13              0 24750    0  [  ] [  ] [  ] [  ]
4     Meeeee2            2 11250    0  [  ] [  ] [  ] [  ]
5 B   deadhead           2 53850    0  [  ] [  ] [  ] [  ]
6     odbphatmackakabig  1  3750 2000  [  ] [  ] [  ] [  ]
                         My cards are: [7s] [7h] [4d] [Ac]
```

**Comment by ctm** [comment]
So, the issue is that `advance_button` pays enough attention to which seat had which blinds last to always (AFAIK!) advance the button properly. It even makes a notation when the small blind is to be omitted, but it doesn't set up enough info for `assign_blinds` to do the right thing.

In theory the separation of responsibility into two different methods was a good idea. In practice, I could have done a better job.

I think I can fix this, but unfortunately, I tried to rush sit 'n' goes out the door and ran into one bug after another in them, none of which showed up in the light testing I did on my development machine. So, this is not going to be fixed for this evening's game :-(
